core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/NamedOperationResolver.java

```diff
@@ -45,25 +52,71 @@
 @JsonPropertyOrder(alphabetic = true)
 public class NamedOperationResolver implements GetFromCacheHook {
     private static final Logger LOGGER = LoggerFactory.getLogger(NamedOperationResolver.class);
+    public static final int TIMEOUT_DEFAULT = 1;
+    public static final TimeUnit TIME_UNIT_DEFAULT = TimeUnit.MINUTES;
     private final NamedOperationCache cache;
+    private final int timeout;
+    private final TimeUnit timeUnit;
+
+    public NamedOperationResolver(final String suffixNamedOperationCacheName) {
+        this(suffixNamedOperationCacheName, TIMEOUT_DEFAULT, TIME_UNIT_DEFAULT);
+    }
 
     @JsonCreator
-    public NamedOperationResolver(@JsonProperty("suffixNamedOperationCacheName") final String suffixNamedOperationCacheName) {
-        this(new NamedOperationCache(suffixNamedOperationCacheName));
+    public NamedOperationResolver(@JsonProperty("suffixNamedOperationCacheName") final String suffixNamedOperationCacheName,
+                                  @JsonProperty("timeout") final int timeout,
+                                  @JsonProperty("timeUnit") final TimeUnit timeUnit) {
```

Comment:
How does this look in json? Would it be easier to just allow timeout in seconds maybe?

Reply:
@GCHQDev404 showed me that the JSON looks like so:
```json
{
  "class" : "uk.gov.gchq.gaffer.graph.hook.NamedOperationResolver",
  "suffixNamedOperationCacheName" : "suffix",
  "timeUnit" : "MINUTES",
  "timeout" : 1
}
```
This is very usable, but this should be documented